src/main/java/net/coco/chess/Board.java

```diff
@@ -1,128 +1,115 @@
-        addOtherWhitePiece(Piece.createWhiteRook());
-        addOtherWhitePiece(Piece.createWhiteKnight());
-        addOtherWhitePiece(Piece.createWhiteBishop());
-        addOtherWhitePiece(Piece.createWhiteQueen());
-        addOtherWhitePiece(Piece.createWhiteKing());
-        addOtherWhitePiece(Piece.createWhiteBishop());
-        addOtherWhitePiece(Piece.createWhiteKnight());
-        addOtherWhitePiece(Piece.createWhiteRook());
+    private String getPiecesToString(List<Piece> pieces) {
+        return pieces.stream()
+                .map(piece -> String.valueOf(piece.getPieceType().getRepresentation(piece.getColor())))
+                .collect(Collectors.joining());
     }
 
-    public String getWhitePawnsResult(String color) {
-        if (color.equals(Piece.WHITE))
-            return getPiecesToString(whitePawns);
-        throw new IllegalArgumentException("매개변수를 확인하세요.");
-    }
+    public void move(String pointStr, Piece piece) {
+        Point point = new Point(pointStr);
 
-    public String getBlackPawnsResult(String color) {
-        if (color.equals(Piece.BLACK))
-            return getPiecesToString(blackPawns);
-        throw new IllegalArgumentException("매개변수를 확인하세요.");
+        ranks.get(point.getColumn()).movePieceToPoint(piece, point.getRow());
+        //todo : piece가 있던 위치도 blank로 변경해야함
     }
 
-    public String getOtherWhitePiecesResult(String color) {
-        if (color.equals(Piece.WHITE))
-            return getPiecesToString(otherWhitePieces.getList());
-        throw new IllegalArgumentException("매개변수를 확인하세요.");
+    public double calculateScore(Color color) {
+        double sum = 0.0;
+        for (int column = 0; column < ranks.size(); column++) {
+            List<Piece> pieces = getPieces(color, column);
+            sum += calculateColumnScore(pieces);
+        }
+        return sum;
     }
 
-    public String getOtherBlackPiecesResult(String color) {
-        if (color.equals(Piece.BLACK))
-            return getPiecesToString(otherBlackPieces);
-        throw new IllegalArgumentException("매개변수를 확인하세요.");
+    private double calculateColumnScore(List<Piece> pieces) {
+        int pawnCount = 0;
+        double sum = 0.0;
+        for (Piece piece : pieces) {
+            pawnCount += piece.getPieceType() == PieceType.PAWN ? 1 : 0;
+            sum += piece.getScore();
+        }
+        sum -= pawnCount > 1 ? pawnCount * 0.5 : 0;
+        return sum;
     }
 
-    private String getPiecesToString(List<Piece> pieces) {
-        return pieces.stream()
-                .map(piece -> String.valueOf(piece.getRepresentation()))
-                .collect(Collectors.joining());
+    private List<Piece> getPieces(Color color, int column) {
+        ArrayList<Piece> getPieces = new ArrayList<>();
+        for (int row = 0; row < ranks.size(); row++) {
+            Rank getRank = ranks.get(row);
+            getPieces.add(getRank.findPieceByColumn(color, column));
+        }
+        return getPieces;
     }
```

Comment:
```suggestion
return ranks.stream()
                .map(getRank -> getRank.findPieceByColumn(color, column))
                .collect(Collectors.toList());
```
스트림을 이용하면 위와 같이 바꿀 수 있습니다 :)
다른곳에도 스트림을 적용할 수 있는 부분이 있는지 찾아본 후 적용해보세요~

Reply:
와! 감사합니다! 적용할 수 있음 다 해보겠습니다!

Reply:
ㅇㅇ 이런 건 스트림 쓰는게 가독성이 훨씬 좋군요.